zerver/migrations/0144_realm_emoji_drop_unique_constraint.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+import os
+import shutil
+import tempfile
+
+from typing import Dict, Optional
+
+from boto.s3.bucket import Bucket
+from boto.s3.connection import S3Connection
+from boto.s3.key import Key
+from django.conf import settings
+from django.db import migrations, models
+from django.db.backends.postgresql_psycopg2.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+class Uploader:
+    def __init__(self) -> None:
+        self.old_orig_image_path_template = "{realm_id}/emoji/{emoji_file_name}.original"
+        self.old_path_template = "{realm_id}/emoji/{emoji_file_name}"
+        self.new_orig_image_path_template = "{realm_id}/emoji/images/{emoji_file_name}.original"
+        self.new_path_template = "{realm_id}/emoji/images/{emoji_file_name}"
```

Comment:
Why do you add the `/images/` to the path here?

Reply:
Storing the files in the same directory can lead to a clash, e.g., there can already an realm emoji with name `123` which will clash with id of some realm emoji. Does it make sense?

Reply:
I see, good thought.  A pretty unlikely issue; I would consider using a temporary intermediate directory and then moving things to the final place you want, since I'd prefer to not need to extend the URLs permanently for this unlikely migration problem.

Reply:
> I would consider using a temporary intermediate directory and then moving things to the final place you want

Sorry I didn't get you here, we will need to change the directory as we can't overwrite the previous images as it will break old references to it, isn't it?

Reply:
Ahh, I see, that's what I was missing.  Yes, you're right.